File: WWSMobile/app/src/main/java/kaufi/jonas/wwsmobile/model/Supplier.kt
package kaufi.jonas.wwsmobile.model

import java.util.*

class Supplier
{
    var id: Int = 0
    var name: String? = null
    var string Place { get; set; }
    var string EMail { get; set; }
    var string Phone { get; set; }
    var string Street { get; set; }
    var string Postcode { get; set; }
    var decimal? SupplierProportion { get; set; }
    var int PickUp { get; set; }
    var DateTime CreationDate { get; set; }
    var string Notes { get; set; }
    var Title Title { get; set; }
    var string Company { get; set; }

    public virtual ICollection<Article> Articles { get; set; }
    public virtual ICollection<ArticleReservation> Reservations { get; set; }

    public override string ToString()
    {
        return Name;
    }
}

[TypeConverter(typeof(EnumDescriptionTypeConverter))]
public enum Title
{
    [Description("Herr")]
    Mr,
    [Description("Frau")]
    Mrs
}

File: Warenwirtschaftssystem/Model/Db/Supplier.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Globalization;

namespace Warenwirtschaftssystem.Model.Db
{
    public class Supplier
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Place { get; set; }
        public string EMail { get; set; }
        public string Phone { get; set; }
        public string Street { get; set; }
        public string Postcode { get; set; }
        public decimal? SupplierProportion { get; set; }
        public int PickUp { get; set; }
        public DateTime CreationDate { get; set; }
        public string Notes { get; set; }
        public Title Title { get; set; }
        public string Company { get; set; }

        public virtual ICollection<Article> Articles { get; set; }
        public virtual ICollection<ArticleReservation> Reservations { get; set; }

        public override string ToString()
        {
            return Name;
        }
    }

    [TypeConverter(typeof(EnumDescriptionTypeConverter))]
    public enum Title
    {
        [Description("Herr")]
        Mr,
        [Description("Frau")]
        Mrs
    }

    static class TitleExtensions
    {
        public static string GetDescription<T>(this T e) where T : IConvertible
        {
            string description = null;

            if (e is Enum)
            {
                Type type = e.GetType();
                Array values = Enum.GetValues(type);

                foreach (int val in values)
                {
                    if (val == e.ToInt32(CultureInfo.InvariantCulture))
                    {
                        var memInfo = type.GetMember(type.GetEnumName(val));
                        var descriptionAttributes = memInfo[0].GetCustomAttributes(typeof(DescriptionAttribute), false);
                        if (descriptionAttributes.Length > 0)
                        {
                            description = ((DescriptionAttribute)descriptionAttributes[0]).Description;
                        }

                        break;
                    }
                }
            }

            return description;
        }
    }

    public class SupplierProportion
    {
        [Key]
        public decimal FromPrice { get; set; }
        public decimal Proportion { get; set; }
    }
}
